In [1]:
import pandas as pd
df=pd.read_csv("custdata.tsv", sep="\t")

In [2]:
df.head()

,custid,sex,is.employed,income,marital.stat,health.ins,housing.type,recent.move,num.vehicles,age,state.of.res
0,2068,F,NaN,11300,Married,True,Homeowner free and clear,False,2.0,49.0,Michigan
1,2073,F,NaN,0,Married,True,Rented,True,3.0,40.0,Florida
2,2848,M,True,4500,Never Married,False,Rented,True,3.0,22.0,Georgia
3,5641,M,True,20000,Never Married,False,Occupied with no rent,False,0.0,22.0,New Mexico
4,6369,F,True,12000,Never Married,True,Rented,True,1.0,31.0,Florida


In [3]:
df.shape

(1000, 11)

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   custid        1000 non-null   int64  
 1   sex           1000 non-null   object 
 2   is.employed   672 non-null    object 
 3   income        1000 non-null   int64  
 4   marital.stat  1000 non-null   object 
 5   health.ins    1000 non-null   bool   
 6   housing.type  944 non-null    object 
 7   recent.move   944 non-null    object 
 8   num.vehicles  944 non-null    float64
 9   age           1000 non-null   float64
 10  state.of.res  1000 non-null   object 
dtypes: bool(1), float64(2), int64(2), object(6)
memory usage: 79.2+ KB


In [5]:
df.describe()

,custid,income,num.vehicles,age
count,1.000000e+03,1000.000000,944.000000,1000.000000
mean,6.984997e+05,53504.771000,1.916314,51.699815
std,4.135083e+05,65478.065729,1.101618,18.863433
min,2.068000e+03,-8700.000000,0.000000,0.000000
25%,3.456668e+05,14600.000000,1.000000,38.000000
50%,6.934030e+05,35000.000000,2.000000,50.000000
75%,1.044606e+06,67000.000000,2.000000,64.000000
max,1.414286e+06,615000.000000,6.000000,146.680197


**Initial Observations**

    - The num.vehicles,housing.type,and recent.move all have a count of 944, but there are 1000 rows, so there may be some missing values. 
    - Also custid is registering as an integer when it may actaully be an object in this case since the mean is not needed here.
    - The is.employed also has missing values since NaN is present and the count is off.
    - The min age is 0 and max age is 146.68 which is not realistic and a possible error.
    - The minimum income is -8700 which seems suspicious for an income.

In [6]:
df.isna().sum()

custid            0
sex               0
is.employed     328
income            0
marital.stat      0
health.ins        0
housing.type     56
recent.move      56
num.vehicles     56
age               0
state.of.res      0
dtype: int64

### Correlations Before Cleaning

In [7]:
df[["age","income", "num.vehicles"]].corr()

,age,income,num.vehicles
age,1.000000,0.027429,-0.063666
income,0.027429,1.000000,0.139198
num.vehicles,-0.063666,0.139198,1.000000


In [8]:
print(df.duplicated().sum())

0


In [9]:
df[df["income"] < 0]


,custid,sex,is.employed,income,marital.stat,health.ins,housing.type,recent.move,num.vehicles,age,state.of.res
691,971703,M,False,-8700,Married,True,Homeowner with mortgage/loan,False,3.0,60.0,Oklahoma


In [10]:
df[df["age"] > 100]

,custid,sex,is.employed,income,marital.stat,health.ins,housing.type,recent.move,num.vehicles,age,state.of.res
212,287882,M,True,60000,Married,True,Rented,True,2.0,137.700030,Florida
266,364991,F,NaN,12000,Divorced/Separated,True,Homeowner with mortgage/loan,False,1.0,123.061023,Nevada
286,397230,F,True,31200,Married,True,Rented,False,2.0,136.052160,Texas
593,843811,F,True,40000,Never Married,True,Homeowner with mortgage/loan,False,1.0,146.680197,Indiana
625,875771,M,False,175200,Married,True,Homeowner with mortgage/loan,False,3.0,124.845087,Arkansas
752,1047534,M,True,200000,Married,True,Homeowner with mortgage/loan,False,4.0,130.401605,Washington
926,1303929,F,False,43400,Never Married,True,Rented,False,1.0,126.737284,West Virginia
953,1361012,F,NaN,255000,Widowed,True,Homeowner free and clear,False,1.0,113.337207,Tennessee


In [11]:
df[df["age"] == 0]

,custid,sex,is.employed,income,marital.stat,health.ins,housing.type,recent.move,num.vehicles,age,state.of.res
667,931449,M,NaN,42400,Married,True,Homeowner free and clear,False,2.0,0.0,Missouri
868,1229417,F,True,48000,Married,True,Homeowner with mortgage/loan,False,3.0,0.0,Washington
965,1375846,F,True,113950,Divorced/Separated,True,Rented,False,2.0,0.0,Louisiana


In [12]:
df.shape

(1000, 11)

In [13]:
df=df.dropna(subset=["num.vehicles"])

In [14]:
df.shape

(944, 11)

In [15]:
df =df[df["income"] >= 0]        

In [16]:
df.shape

(943, 11)

In [17]:
df =df[df["age"]> 0]

In [18]:
df.shape

(940, 11)

In [19]:
df =df[df["age"]<= 115]

In [20]:
df.shape

(933, 11)

### Correlations After Cleaning

In [21]:
df[["age","income", "num.vehicles"]].corr()

,age,income,num.vehicles
age,1.000000,0.009160,-0.069044
income,0.009160,1.000000,0.134475
num.vehicles,-0.069044,0.134475,1.000000


**Interpertation:** After cleaning the data income and number of vehicles had the highest correlation with a Pearson's r value of 0.134. Although it is the highest, it is a weak positive relationship. Age and income had almost no linear relationship at all with a r value of 0.009160 and age and number of vehicles had a very weak negative relationship with a r-value of -0.069.

**Cleaning Log**

- There was a total of 1000 rows loaded. I decided to drop the 56 rows which had data missing for num.vehicles so all 3 correlations used complete rows.
- I excluded 1 row that had a negative income as negative income in this case is invalid.
- I excluded 3 rows that had an age of 0 because this is unrealistic as they are also listed as married and divorced. 
- I excluded 7 rows that had an age greater than 115 as these ages seem unrealistic. I chose 115 as my upper age cutoff for this analysis. 
- In total 67 rows were excluded from the analysis. 

https://github.com/jazzminowens/cs82a-portfolio